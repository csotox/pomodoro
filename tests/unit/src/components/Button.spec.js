import { shallowMount } from '@vue/test-utils';
import Button from '@/components/Button.vue';

function createButtonView() {
  return shallowMount(Button, {
    slots: {
      default: 'Haz tu primer pomodoro',
    },
  });
}

describe('Button.vue', () => {
  it('should render properly', () => {
    const button = createButtonView();
    expect(button).toMatchSnapshot();
  });

  describe('Methods', () => {
    describe('onClick', () => {
      it('should emit an event named click', () => {
        const button = createButtonView();

        button.find('.button').trigger('click');

        expect(button.emitted('click')).toBeTruthy();
      });
    });
  });
});
