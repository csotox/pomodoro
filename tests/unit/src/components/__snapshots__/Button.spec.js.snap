// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.vue should render properly 1`] = `<button class="button">Haz tu primer pomodoro</button>`;
